src/test/java/com/synopsys/integration/jenkins/detect/substeps/DetectExecutionManagerTest.java

```diff
@@ -0,0 +1,57 @@
+package com.synopsys.integration.jenkins.detect.substeps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jenkins.detect.exception.DetectJenkinsException;
+
+public class DetectExecutionManagerTest {
+
+    private static DetectJarManager detectJarManager = new DetectJarManager(null, null, null, null);
+    private static DetectJarManager spiedDetectJarManager = Mockito.spy(detectJarManager);
+
+    @Test
+    public void testCallInterruptedException() {
+        try {
+            Mockito.doThrow(new InterruptedException()).when(spiedDetectJarManager).setUpForExecution();
+        } catch (IOException | InterruptedException e) {
+            fail("Unexpected exception thrown", e);
+        }
```

Comment:
Rather than mucking around with Mockito you can quite easily accomplish the same behavior with an inline class like so:

```
        DetectExecutionManager detectExecutionManager = new DetectExecutionManager() {
            @Override
            public DetectSetupResponse setUpForExecution() throws IOException, InterruptedException, IntegrationException {
                throw new InterruptedException();
            }
        };
```

It's the same number of lines and I think it's a bit clearer what you're doing.

Reply:
This is MUCH clearer and cleaner.